File: PRAKTIKUM OOP/3rd_week/Rekening.cpp
#include <iostream>
#include "Rekening.h"
using namespace std;

Rekening::Rekening(double value) {
    this->saldo = value>0? value : 0;
}

void Rekening::setSaldo(double value) {
    this->saldo = value;
}
double Rekening::getSaldo() const {
    return this->saldo;
}

void Rekening::simpanUang(double value) {
    this->saldo += value;
} 

bool Rekening::tarikUang(double value) {
    if (value <= this->saldo) {
        this->saldo -= value;
        return true;
    }
    return false;
}

File: PRAKTIKUM OOP/LatihanUAS/batch1/Rekening.java
public abstract class Rekening {
    protected String nama;
    protected double saldo;
    protected double sukuBunga;

    public Rekening(String nama, double saldo, double sukuBunga) {
        this.nama = nama;
        this.saldo = saldo;
        this.sukuBunga = saldo < 0 ? 0 : sukuBunga;
    }

    public abstract void setor(double value);
    
    public abstract void tarik(double value);

    public abstract void update();

    public double hitungBiaya() {
        return Math.min(0.1 * this.saldo, 10);
    }

    public String getNama() {
        return nama;
    }

    public double getSaldo() {
        return saldo;
    }

    public double getSukuBunga() {
        return sukuBunga;
    }
}